src/scene/triangle.h

```diff
@@ -2,19 +2,22 @@
 #ifndef __TRIANGLE_H__
 #define __TRIANGLE_H__
 
+#include <glm/glm.hpp>
+
 #include "raycast/hit.h"
 #include "raycast/ray.h"
-#include <glm/glm.hpp>
 
 class Triangle {
   glm::vec3 mP0, mP1, mP2;
   glm::vec3 mNormal;
   glm::vec3 mColor;
 
+  const unsigned int mId;
+
 public:
-  Triangle(glm::vec3 p0, glm::vec3 p1, glm::vec3 p2);
+  Triangle(glm::vec3 p0, glm::vec3 p1, glm::vec3 p2, unsigned int id);
```

Comment:
If we are going to have id's they likely need to be unique.  May want to instead have id be assigned in the constructor by calling a static genId() function.